lib/World.luau

```diff
@@ -1012,11 +1011,7 @@ end
 function World:insert(id, ...)
 	debug.profilebegin("insert")
 
-	assertComponentsProvided(...)
-
-	if not self:contains(id) then
-		error(ERROR_NO_ENTITY, 2)
-	end
+	assertWorldOperationIsValid(self, id, ...)
```

Comment:
I think we should separate between checking components and component instances and remove the assertions that are happening in the loop on e.g. line 1022 under world:insert but also places like world:replace, remove etc

Reply:
Are there any concerns to be had here about increasing the amount of iterations just to split out the assertions? 

The `assertValidComponentInstance` and `assertValidComponent` assert functions are often called in loops first, and after that other things are then done. We will be iterating over the component arguments once to assert, and then again in the methods themselves.

I would assume the impact is negligible though.

Reply:
Hm it would be negligible but you are right, there is not much of a reason to refactor it now if it works.